### PR TITLE
Cart

### PR DESCRIPTION
**Pull request by lannisterrr**
Added : full logic of add to cart
Added : Increase and decrease the count of the item

### CODE REVIEW

File: src/components/CartItem.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  const { dispatch } = useCart();
+  return (
+    <section class="cart__card-container box-shadow-right">
+      <div class="cart__img-div">
+        <img src={image} alt={productName} class="cart-img" />
+      </div>
+      <div class="cart__text-div p-6">
+        <h2>{name}</h2>
+        <p className="l-h-3">
+          <span className="price f-7">Rs. {parseInt(price)}</span>
```
Comment (GuptaEesh) [inline src/components/CartItem.js:15]: You shouldn't use parseInt here as the ideal purpose is to extract number out of a string.
Number() should be used for conversion into integers.
> Extracting and Converting are distinct things.

File: src/Styles/cart.css [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+/* buttons */
+
+.CartItem-count {
+  padding: 5px 10px;
+  border: 1px solid #ccc;
+}
+.CartItem-addOne,
+.CartItem-removeOne {
+  padding: 5px 10px;
+  border: 1px solid #ccc;
```
Comment (GuptaEesh) [inline src/Styles/cart.css:76]: Why not make a variable naming for ecomm separately? I think it will help.

File: src/pages/Cart.js [hunk 3]
```diff
@@ -1,5 +1,29 @@
 import React from 'react';
+import CartItem from '../components/CartItem';
```
Comment (GuptaEesh) [inline src/pages/Cart.js:2]: why not do barrel import.? 
It would wind up your imports in a single line.

**Review by GuptaEesh — COMMENTED** [review]
>Added some suggestions. 
Other that that the code looks good.

>UseReducer use can be seen which is great for handling multiple operations.

>

File: src/context/cart-context.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  cartItems: 0,
+  totalPrice: 0,
+};
+
+const CartProvider = ({ children }) => {
+  const [state, dispatch] = useReducer(cartReducer, initialState);
+
+  let cartData = displayCart(state, state.cart);
+
+  console.log('cart: ', state.cart);
```
Comment (AnuragSahu11) [inline src/context/cart-context.js:18]: You can remove console.log

File: src/utils/cart-utils.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+        ...state,
+        cart: [...state.cart, action.payload],
+        cartItems: !existingItem ? state.cartItems + 1 : state.cartItems,
+        totalPrice: state.totalPrice + action.payload.price,
+      };
+    case 'REMOVE_FROM_CART':
+      let index = state.cart.findIndex(i => i.id === action.payload.id); // the index where item is more than 1
+      const copy = [...state.cart];
+      copy.splice(index, 1);
+      console.log('copy', copy);
```
Comment (AnuragSahu11) [inline src/utils/cart-utils.js:15]: console.log can be removed

**Review by AnuragSahu11 — APPROVED** [review]
-Proper naming conventions followed
-Good use of array functions
-Good work separating each components in it's own file
You can merge

**Pull request merged**